### PR TITLE
CCPoint extension

### PR DESCRIPTION
**Pull request by Klaim**
Some additional basic functions to ease manipulation of CCPoint.

### CODE REVIEW

File: cocos2dx/include/CCGeometry.h [hunk 1]
```diff
@@ -46,6 +48,93 @@ class CC_DLL CCPoint
...
+}
+    
+inline CCPoint operator*( const CCPoint a, const float factor )
+{
+    return CCPoint( a.x * factor, a.y * factor );
+}
+
+inline CCPoint operator/( const CCPoint a, const float factor )
+{
+    return CCPoint( a.x / factor, a.y / factor );
```
Comment (minggo) [inline cocos2dx/include/CCGeometry.h:80]: How about add 

```
assert(factor != 0)
```

before return?

File: cocos2dx/include/CCGeometry.h [hunk 2]
```diff
@@ -46,6 +48,93 @@ class CC_DLL CCPoint
...
+
+inline CCPoint& operator*=( CCPoint& a, const float factor )
+{
+    a = a * factor;
+    return a;
+}
+
+inline CCPoint& operator/=( CCPoint& a, const float factor )
+{
+    a = a / factor;
```
Comment (minggo) [inline cocos2dx/include/CCGeometry.h:104]: How about add 

```
assert(factor != 0)
```

before doing dividing?

Comment (Klaim) [inline cocos2dx/include/CCGeometry.h:104]: Right, I'll add it.

File: cocos2dx/include/CCGeometry.h (hunk 1 above)
Comment (Klaim) [inline cocos2dx/include/CCGeometry.h:80]: Right.

**Comment by minggo** [comment]
Thank you.
And i will add you into AUTHORS.

**Pull request merged**